### PR TITLE
POC: Editor: Multipage Canvas

### PR DESCRIPTION
**Pull request by barklund**
## Context

This adds a preview of the pages before and after the current ones to either side of the canvas, if there is enough space. It adds as many pages, as there is room for.

## To-do

- [x] Which size should they pages be and with what margin?
- [x] Should the pages be active in any way? Can you click them? Can you drag stuff to them? Or from them?

## User-facing changes

| Window height | Screenshot |
|-|-|
| Narrow | Screen Shot 2022-06-10 at 17 32 42 |
| Medium | Screen Shot 2022-06-10 at 17 31 52 |
| Tall | Screen Shot 2022-06-10 at 17 32 15 |

It even works in RTL:
Screen Shot 2022-06-10 at 17 40 17

## Testing Instructions

This PR can be tested by following these steps:

0. Enable _"Context pages"_ feature flag
1. Add multiple pages to a story (preferably with some content
2. Switch page a few times
3. Observe the previous and future pages in the margins of the current page
4. Try different browser sizes to see the changes

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #9643

### CODE REVIEW

**Comment by timarney** [comment]
Nice -- overall :)

Seeing some issues with previous and next screen previews --- likely separate from this PR as the "thumbnail" view has same issue. 
Screen Shot 2022-06-10 at 1 33 19 PM

**Review by sblinde — APPROVED** [review]
🐝 

Code looks solid. Do agree: there are some pre-existing issues w/ the thumbnail view of pages where the `Fit` button overlaps the arrows when there's a lot of pages, but I know that's outside of this scope and body of work.

**Comment by swissspidy** [comment]
Are the pages basically all just screenshots? What is the performance impact of this? Can we measure it?

It might be best to put this behind a feature flag until that is more clear.

**Comment by barklund** [comment]
> Are the pages basically all just screenshots? What is the performance impact of this? Can we measure it?

They are all instances of `PagePreview`, so they're complex HTML until they've been snapshotted by the snapshot logic that we have in place. There might very well be a performance impact of this, as we previously only snapshot tiny thumbnails in the carousel, but now snapshot these much larger canvases. It will have an impact on both computation time for generating these snapshots as well as memory used for storing them. I have not putting any tracking of that in at this time.

> It might be best to put this behind a feature flag until that is more clear.

That sounds like a good idea!

**Comment by merapi** [comment]
I'm trying to make it work on QA so the design pod can review it, right now there is no build ZIP.

File: packages/story-editor/src/components/canvas/extraPages.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        listWidth={listWidth}
+        extraPageHeight={extraPageHeight}
+      >
+        {pagesToShow.map((pageNum) => (
+          <ExtraPage key={pageNum} extraPageWidth={extraPageWidth}>
+            <ExtraPagePreview
+              page={pages[pageNum]}
+              onClick={clickPage(pages[pageNum].id)}
+              aria-label={sprintf(
+                /* translators: %s: Page number */
```
Comment (swissspidy) [inline packages/story-editor/src/components/canvas/extraPages.js]:
```suggestion
                /* translators: %s: page number. */
```

To address the i18n lint warning

> Warning: The string "Go to page %s" has 2 different translator comments.

**Comment by merapi** [comment]
I can reproduce the bug from 00:13 locally and on QA.

https://user-images.githubusercontent.com/21036224/175024042-4ec2c652-2610-414e-89d4-7af134c137ce.mov

Other than that we are waiting for the design pod review (page size/margins etc).

**Comment by barklund** [comment]
> I can reproduce the bug from 00:13 locally and on QA.

Whoa, that's a weird bug! I'll investigate, thanks for catching it!

**Comment by barklund** [comment]
Approved by UX:
- Size, padding, opacity, and hover looks good

To do after UX review:
- [ ] Last page never shows (some off-by-one error somewhere)
- [ ] Resize all page previews to also show non-safe space (includes page carousel and grid view, should probably be new ticket)
- [ ] Don't show hover state when dragging from canvas to neighbor page
- [ ] Don't allow neighbor pages to be draggable
- [ ] Measure impact on low-end devices
- [ ] Fade pages even more further away, show max 4 either direction

**Pull request merged**